### PR TITLE
feat: add VariableMappings.fromVariablesInOtherTables

### PR DESCRIPTION
**Pull request by BrendaHijmans**
Adds the ability to map from different tables. Needs changes in variable explorer.

### CODE REVIEW

**Comment by mswertz** [comment]
looks good. Would it help if I try to get the UX in line with this?

**Comment by BrendaHijmans** [comment]
> looks good. Would it help if I try to get the UX in line with this?

yes!

File: apps/catalogue/src/components/HarmonizationDefinition.vue [hunk 1]
```diff
@@ -7,17 +7,17 @@
         <dd class="col-10">
           <template
             v-if="variable.cohortMapping && variable.cohortMapping.description"
-            >{{ variable.cohortMapping.description }}</template
```
Comment (connoratrug) [inline apps/catalogue/src/components/HarmonizationDefinition.vue:10]: white space removed by accident ?

File: apps/catalogue/src/views/VariableView.vue [hunk 2]
```diff
@@ -149,6 +149,7 @@
             <tr>
               <th>fromTable</th>
               <th>fromVariables</th>
+              <th>fromVariablesInOtherTables</th>
```
Comment (connoratrug) [inline apps/catalogue/src/views/VariableView.vue:152]: The whole things seems confusing to me. By trying to force a ono-to-one mapping of model and interface the storage details bleed into the user interface. Suggest writing down who is using the screen to accomplish what tasks. Then setup the interface to support this task. Instead to visualising the storage model.

File: apps/catalogue/src/components/HarmonizationDefinition.vue (hunk 1 above)
Comment (mswertz) [reply apps/catalogue/src/components/HarmonizationDefinition.vue:10]: I think we should agree on formatter. I saw similar things in your pull request

File: apps/catalogue/src/views/VariableView.vue (hunk 2 above)
Comment (mswertz) [reply apps/catalogue/src/views/VariableView.vue:152]: in this case we could do this indeed. You may have noticed that in other parts of the UI I indeed merged them. Good suggestion to do that here too.

Comment (mswertz) [reply apps/catalogue/src/views/VariableView.vue:152]: NB, I disagree with the overall principele that there shouldn't be one-to-one mapping of model and interface. This is scientific software and there the data submitters are just (or even more) important then the users. Hence a much higher transparancy is required in terms on what actually is being stored (as compared to consumer software were indeed you should often hide the underlying model)

Comment (mswertz) [reply apps/catalogue/src/views/VariableView.vue]: done

**Review by mswertz — APPROVED** [review]

**Pull request merged**